test-bins/src/shutdown.rs

```diff
@@ -0,0 +1,35 @@
+use std::io;
+
+use log::info;
+use tokio::{signal, signal::unix::SignalKind};
+
+pub struct Shutdown;
+impl Shutdown {
+    pub async fn wait() -> Result<(), io::Error> {
```

Comment:
nit: can't we just have a single function, without extra calls? Just having two `wait` methods with different cfgs?